packages/next-server/lib/router/router.ts

```diff
@@ -280,11 +285,8 @@ export default class Router implements BaseRouter {
         method = 'replaceState'
       }
 
-      // @ts-ignore pathname is always a string
-      const route = toRoute(pathname)
-      const { shallow = false } = options
-
       if (isDynamicRoute(route)) {
+        this.parsedDynamicRoute = true
```

Comment:
This seems like a strange state 🤔 It'll not be reset after the first time it happens for subsequent route changes it'll always be true

Reply:
Changed how it's being handled to move the dynamic values parsing above the hash change check so this is no longer used.